File: parking_system/Spot.java
package parking_system;

class Spot {
    private String spotId;
    private Vehicle vehicle;

    public Spot(String spotId) {
        this.spotId = spotId;
    }

    public boolean isFree() {
        return vehicle == null;
    }

    public void setOccupied(Vehicle veh) {
        vehicle = veh;
    }

    public void setFree() {
        this.vehicle = null;
    }

    @Override
    public String toString() {
    return "[" + spotId + " -> " + (vehicle != null ? vehicle : "FREE") + "]";
    }
}

File: cpp/parking_system/spot.cpp
#include "spot.h"
#include <string>

using namespace std;

Spot::Spot(const string& spotId) : spotId(spotId), vehicle(nullptr) {}

string Spot::to_string() const {
    string res = spotId;
    res += " ";
    res += (!vehicle) ? "NULL" : vehicle->to_string();
    return res;
}

bool Spot::isVacant() const {
    return vehicle == nullptr;
}

void Spot::setOccupied(Vehicle* v) {
    vehicle = v;
}

void Spot::setFree() {
    vehicle = nullptr;
}
